kong/kong.lua

```diff
@@ -252,6 +252,18 @@ function Kong.balancer()
   end
 end
 
+function Kong.rewrite()
+  core.rewrite.before()
+
+  for _, plugin in ipairs(singletons.loaded_plugins) do
+    if plugin.schema.no_consumer and plugin.schema.no_api then
+      plugin.handler:rewrite()
+    end
+  end
+
+  core.rewrite.after()
+end
+
```

Comment:
sorry my bad, `loaded_plugins` are the ones actively configured in the DB. So that is ok.

But then why the `if plugin.schema.no_consumer and plugin.schema.no_api then`? If the plugin is using the rewrite phase, then by definition it must be a global plugin (no consumer and no api), so why the extra flag `no_api`?

Also: because it is based on the database contents, it requires reloading on invalidation.